opentracing-util/src/main/java/io/opentracing/util/TracerSupplier.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2016-2018 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.util;
+
+import io.opentracing.Tracer;
+
+/**
+ * Functional interface for {@code Supplier<Tracer>} as the API is still Java 1.6+ compatible.
+ */
+public interface TracerSupplier {
```

Comment:
Is this interface required to resolve #186 ?

Reply:
If you want to use it with lambda's while maintaining jdk 1.6 compatibility, sadly I think it is.

Reply:
Making it "lambda" like is not required to resolve #186. If there is no requirement for it I am -1 on adding this interface.

Reply:
Ok, noted. However obtaining and configuring a tracer is potentially expensive / complex behaviour that may not be necessary. This is -in my opinion- a good enough reason to introduce the method _here_.
How do we reach consensus on this one; let people vote?

Reply:
I would actually be in favor of **only** accepting TracerSupplier (btw, isn't "provider" a more accepted term?), because the semantics of "if absent" imply the tracer may not be needed and accepting an actual tracer instance in that case is promoting a bad pattern of constructing an (expensive) instance that may not be needed. E.g. the correct behavior of the caller if we allow Tracer would be to do something like
```
Tracer tracer  = ... // create
if (!GlobalTracer.registerIfAbsent(tracer)) {
  tracer.close();  // release resources
}
```
With "provider" pattern the above is not necessary.

Reply:
I think you are talking about lazy tracer initialization. Then it makes sense

Reply:
btw we can still keep the resolver use case as one liner if it implements a new method
```
GlobalTracer.registerIfAbsent(Tracerresolver.provider());
```

Reply:
I have no problems removing the `registerIfAbsent(Tracer)` method, as it can be replaced with the trivial supplier inner-class (as the current delegated implementation) or a lambda in somewhat decent jvm version.

Reply:
@yurishkuro I thought of `provider`, but chose `TracerSupplier` as the Java 8 functional interface would be a `Supplier<Tracer>`. This is also why I chose the name for the `get()` method: That's what the `Supplier<T>` has in java 8.

Reply:
I'm not very biased to either choice so if somebody has strong feelings about the one or the other, that's okay by me.

Reply:
@pavolloffay @yurishkuro Re: `TracerSupplier`
>Is this interface required to resolve #186 ?

Come to think of it, technically, I think we could also get away with `Callable<Tracer>`, which is available in Java 6. The trivial new interface doesn't really bother me, but perhaps you have strong opinions on this?

Using a `Callable<Tracer>` as provider does have a slight disadvantage that we will have to catch and handle the thrown `Exception`

Reply:
Interface `TracerSupplier` has been replaced by `Callable<Tracer>`